expression/builtin_time_vec.go

```diff
@@ -1116,11 +1116,56 @@ func (b *builtinToDaysSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinDateFormatSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinDateFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		if err := handleInvalidTimeError(b.ctx, err); err != nil {
+			return err
+		}
+	}
+
+	buf0, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[1].VecEvalString(b.ctx, input, buf0); err != nil {
+		return err
+	}
+
+	result.ReserveString(n)
+	ds := buf.Times()
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		if ds[i].InvalidZero() {
+			if err := handleInvalidTimeError(b.ctx, types.ErrIncorrectDatetimeValue.GenWithStackByArgs(ds[i].String())); err != nil {
+				return err
+			}
+			result.AppendNull()
+			continue
+		}
+		if buf0.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
```

Comment:
I think the logic is different from [non-vec function](https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L790-L815).
1. check the ```buf.IsNull(i)``` and ```buf0.IsNull(i)```
2. check ```buf0.GetString(i) == ""```
3. check the ```ds[i].InvalidZero()``` and do something like the [non-vec function](https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L804-L815)